pkg/madmin/api-log.go

```diff
@@ -0,0 +1,78 @@
+/*
+ * MinIO Cloud Storage, (C) 2019 MinIO, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package madmin
+
+import (
+	"encoding/json"
+	"net/http"
+	"net/url"
+	"strconv"
+
+	"github.com/minio/minio/cmd/logger/message/log"
+)
+
+// LogInfo holds console log messages
+type LogInfo struct {
+	log.Entry
+	NodeName string `json:"node"`
+	Err      error  `json:"-"`
+}
+
+// GetLogs - listen on console log messages.
+func (adm AdminClient) GetLogs(node string, lineCnt int, doneCh <-chan struct{}) <-chan LogInfo {
+	logCh := make(chan LogInfo, 1)
+	urlValues := make(url.Values)
+	urlValues.Set("node", node)
+	urlValues.Set("limit", strconv.Itoa(lineCnt))
+	// Only success, start a routine to start reading line by line.
+	go func(logCh chan<- LogInfo) {
+		defer close(logCh)
+		for {
+			reqData := requestData{
+				relPath:     "/v1/log",
+				queryValues: urlValues,
+			}
+			// Execute GET to call log handler
+			resp, err := adm.executeMethod("GET", reqData)
+			if err != nil {
```

Comment:
Why not also returning this error to the client ? (like server unreachable or port closed)